### PR TITLE
PROB added

### PR DESCRIPTION
**Pull request by yww211**

### CODE REVIEW

**Comment by deadpoppy** [comment]
I suggest adding a presentation of the results and a citation to our paper to the README
https://www.semanticscholar.org/paper/Weakly-Supervised-Lesion-Localization-With-Pooling/c73e3ee16fe603775aed1341cc6ba3741a7c7ecc

File: config/example_PROB.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+    "train_csv": "/nas/public/CheXpert/CheXpert-v1.0/train.csv",
+    "dev_csv": "/nas/public/CheXpert/CheXpert-v1.0/valid.csv",
```
Comment (deadpoppy) [inline config/example_PROB.json:3]: It is recommended to keep the path anonymous

File: README.md [hunk 2]
```diff
@@ -64,8 +64,13 @@ we test it on 200 patients dataset, got the **AUC** as below:
 |AVG_MAX|No|
 |AVG_MAX_LSE|No|
 
-> We are now implementing heatmap options for `AVG_MAX`, `AVG_MAX_LSE`, and `PROB`, So before that you can use the supported `global_pool` options to train a model and plot the heatmaps.
+* We also provide heatmap comparision here, including AVG, [LSE](https://arxiv.org/abs/1705.02315), and our own PROB pooling.
 
+||&emsp;&emsp;&emsp;&emsp; original &emsp;&emsp;&emsp;&emsp; [AVG (dev mAUC:0.895)](config/example_AVG.json) &emsp; [LSE (dev mAUC:0.896)](config/example_LSE.json) &emsp; [PROB (dev mAUC:0.896)](config/example_PROB.json)|
+|------|-----|
+|Cardiomegaly|<img src="Cardi.jpg" width="100%" align="middle"/>|
```
Comment (yil8) [inline README.md:71]: The heatmap looks so weird here, like the Atelectasis and Consolidation, why is it like bounding box? It feels some bbox annotation is mixed with the training. @yww211 @deadpoppy

Comment (yww211) [reply README.md:71]: > The heatmap looks so weird here, like the Atelectasis and Consolidation, why is it like bounding box? It feels some bbox annotation is mixed with the training. @yww211 @deadpoppy

That is just a coincidence, more examples do not have this pattern.

File: README.md [hunk 3]
```diff
@@ -64,8 +64,14 @@ we test it on 200 patients dataset, got the **AUC** as below:
 |AVG_MAX|No|
 |AVG_MAX_LSE|No|
 
-> We are now implementing heatmap options for `AVG_MAX`, `AVG_MAX_LSE`, and `PROB`, So before that you can use the supported `global_pool` options to train a model and plot the heatmaps.
+* We also provide heatmap comparision here, including AVG, [LSE](https://arxiv.org/abs/1705.02315), and our own PROB pooling.
 
+||&emsp;&emsp;&emsp;&emsp; original &emsp;&emsp;&emsp;&emsp; [AVG (dev mAUC:0.895)](config/example_AVG.json) &emsp; [LSE (dev mAUC:0.896)](config/example_LSE.json) &emsp; [PROB (dev mAUC:0.896)](config/example_PROB.json)|
+|------|-----|
```
Comment (deadpoppy) [inline README.md:70]: The header here is misplaced when it is displayed, and the table needs to be merged and centered. The Markdown syntax may not be supported, so it is recommended to use HTMAL. Here I have simply written one, you can click edit and copy to use it
`<table>
  <tr>
    <td> </td>
		<td>original</td>
		<td>AVG (dev mAUC:0.895)</td>
    <td>LSE (dev mAUC:0.896)</td>
    <td> PROB (dev mAUC:0.896)</td>
	<tr>
  <tr>
    <td>Cardiomegaly </td>
		<td colspan="4"></td>
	<tr>
  <tr>
    <td>Atelectasis </td>
		<td colspan="4"></td>
        <tr>
    <td>Cardiomegaly </td>
		<td colspan="4"></td>
           <tr>
    <td>Consolidation </td>
		<td colspan="4"></td>
</table>`

**Comment by deadpoppy** [comment]
It seems that the pictures you Markdown display are placed in the first level directory. If the following materials are placed here, the directory will be very large. It is recommended to create a new folder to place them in.

File: README.md (hunk 3 above)
Comment (yww211) [reply README.md]: Done.

**Comment by yww211** [comment]
> It seems that the pictures you Markdown display are placed in the first level directory. If the following materials are placed here, the directory will be very large. It is recommended to create a new folder to place them in. @yww211
I think there is no need to change and I would rather to stay the same. Here is a reference, you can check the materials in this repo. https://github.com/jfhealthcare/object-CXR @deadpoppy

File: README.md [hunk 4]
```diff
@@ -64,7 +64,31 @@ we test it on 200 patients dataset, got the **AUC** as below:
...
+  <tr>
+    <td>Atelectasis </td>
+		<td colspan="4"><img src="Atelectasis.jpg"></td>
+        <tr>
+    <td>Pleural Effusion </td>
+		<td colspan="4"><img src="Pleural_Effusion.jpg"></td>
+           <tr>
+    <td>Consolidation </td>
+		<td colspan="4"><img src="Consolidation.jpg"></td>
+</table>`
```
Comment (deadpoppy) [inline README.md:91]: What is the function of the extra point ` here? If it is not special, it is recommended to delete.

Comment (yww211) [reply README.md]: done.

**Comment by deadpoppy** [comment]
> > It seems that the pictures you Markdown display are placed in the first level directory. If the following materials are placed here, the directory will be very large. It is recommended to create a new folder to place them in. @yww211
> > I think there is no need to change and I would rather stay the same. Here is a reference, you can check the materials in this repo. https://github.com/jfhealthcare/object-CXR @deadpoppy

It is still suggested that the code base should put these exhibition picture materials in a separate folder. What you mentioned above is the exhibition website of the competition, and referred to mmdetection. https://github.com/open-mmlab/mmdetection @yww211 @yil8

**Comment by deadpoppy** [comment]
Others have LSTM @yww211

**Comment by yww211** [comment]
> > > It seems that the pictures you Markdown display are placed in the first level directory. If the following materials are placed here, the directory will be very large. It is recommended to create a new folder to place them in. @yww211
> > > I think there is no need to change and I would rather stay the same. Here is a reference, you can check the materials in this repo. https://github.com/jfhealthcare/object-CXR @deadpoppy
> 
> It is still suggested that the code base should put these exhibition picture materials in a separate folder. What you mentioned above is the exhibition website of the competition, and referred to mmdetection. https://github.com/open-mmlab/mmdetection @yww211 @yil8
No need to waste time talking about this meaningless point. **Neither ** pictures in or not in a folder  will have a substantial  impact on this project. I still don't want to change this. @deadpoppy @yil8

**Comment by yww211** [comment]
> Others have LSTM @yww211
LSTM ->LGTM

File: README.md [hunk 5]
```diff
@@ -64,7 +64,31 @@ we test it on 200 patients dataset, got the **AUC** as below:
 |AVG_MAX|No|
 |AVG_MAX_LSE|No|
```
Comment (yil8) [inline README.md:65]: add |PCAM|Yes|

**Comment by yil8** [comment]
@yww211 @deadpoppy I suggest rename `PROB` to `PCAM` in both code and README, as we are going to provide arxiv link to `PCAM Pooling` soon

File: README.md (hunk 5 above)
Comment (yww211) [reply README.md:66]: Done

**Comment by yww211** [comment]
> @yww211 @deadpoppy I suggest rename `PROB` to `PCAM` in both code and README, as we are going to provide arxiv link to `PCAM Pooling` soon

Done

**Pull request merged**